include/heap.h

```diff
@@ -3,81 +3,105 @@
    See LICENSE and COPYING for usage.
 
  *********************************************************************/
+#define MAX_BLOCK_SIZE 1600
+#define MAX_BLOCK_COUNT 16
 
 #define DECLARE_HEAP(type, orderby) \
     struct heap_ ## type {   \
         uint32_t size;    \
         uint32_t n;       \
-        type *top;        \
+        type *top[MAX_BLOCK_COUNT];        \
     }; \
     typedef struct heap_ ## type heap_ ## type; \
+    static inline type* getElement(struct heap_ ## type *heap, uint32_t idx) \
+    { \
+        uint32_t elements_per_block = MAX_BLOCK_SIZE/sizeof(type); \
```

Comment:
Looking at the implementation on the left, we only had divisions by 2 (aka bitshifts), with these divisions you could very well not be bitshifting anymore, making the implementation heavier. Is there a way to go around this? (maybe it's not a bad thing at all)